libpod/container_internal.go

```diff
@@ -951,6 +952,33 @@ func (c *Container) generateSpec() (*spec.Spec, error) {
 		g.SetProcessGID(gid)
 	}
 
+	// Add addition groups if c.config.GroupAdd is not empty
+	if len(c.config.GroupAdd) > 0 {
+		if !c.state.Mounted {
+			return nil, errors.Wrapf(ErrCtrStateInvalid, "container %s must be mounted in order to add additional groups", c.ID())
+		}
+		for _, group := range c.config.GroupAdd {
+			gid64, err := strconv.ParseUint(group, 10, 32)
+			if err != nil {
+				_, gid32, err := chrootuser.GetUser(c.state.Mountpoint, strconv.Itoa(int(g.Spec().Process.User.UID))+":"+group)
```

Comment:
This does not work, I don't think.  If I do
--group-add systemd-coredump

What happens?
I think you need to create a new function in chrootuser called GetGroup that reads from /etc/group

Reply:
Agree, we should be fetching this from /etc/group instead

Reply:
It is being fetched from /etc/group inside GetUser at https://github.com/projectatomic/libpod/blob/master/pkg/chrootuser/user.go#L58.
If the function name `GetUser` is misleading/confusing I can create a `GetGroup` function that does the same thing - although I don't see why have 2 functions basically do the same thing. Or we could rename `GetUser` to `GetUserAndGroup` since it returns both uid and gid for the userspec passed in.